src/marshmallow/schema.py

```diff
@@ -682,7 +682,12 @@ def _deserialize(
                 for key in set(data) - fields:
                     value = data[key]
                     if unknown == INCLUDE:
-                        set_value(typing.cast(typing.Dict, ret), key, value)
+                        set_value(
+                            typing.cast(typing.Dict, ret),
+                            key,
+                            value,
+                            allow_dotted=False,
```

Comment:
I would bypass `set_value()` and just make the dict assignment like @adyouri [suggested](https://github.com/marshmallow-code/marshmallow/issues/1506#issuecomment-583871756) to minimize the complexity.

Reply:
Ah, that sounds so much better! I feel silly now. 😅 

I'm doing that now, and I'm going to rename to `ret_l` and `ret_d` for the `many=True/False` branches, so that mypy is pacified and we don't need `typing.cast(typing.Dict, ret)[key] = value` (which works, but yuck...).